examples/Example17/src/PrimaryGeneratorMessenger.cc

```diff
@@ -45,6 +49,44 @@ PrimaryGeneratorMessenger::PrimaryGeneratorMessenger(PrimaryGeneratorAction *Gun
   fRndmDirCmd->SetRange("rBeamDir>=0.&&rBeamDir<=1.");
   fRndmDirCmd->AvailableForStates(G4State_Idle);
 
+  fRandomizeGunCmd = new G4UIcmdWithABool("/example17/gun/randomizeGun", this);
+  fRandomizeGunCmd->SetGuidance("Shoot particles in random directions within defined Phi and Theta ranges, the "
+                                "particle type is also selected at random from the selected options");
+  fRandomizeGunCmd->AvailableForStates(G4State_Idle);
+
+  fAddParticleCmd = new G4UIcmdWithAString("/example17/gun/addParticle", this);
+  fAddParticleCmd->SetGuidance("When using randomization, add a particle to the list of possibilities, with the option "
+                               "to provide a weight to indicate how often it should appear");
+  fAddParticleCmd->SetParameterName("rParticleName", false);
+  fAddParticleCmd->SetDefaultValue("geantino");
+
+  fMinPhiCmd = new G4UIcmdWithADoubleAndUnit("/example17/gun/minPhi", this);
+  fMinPhiCmd->SetGuidance("Minimum phi angle when using randomization, units deg or rad");
+  fMinPhiCmd->SetParameterName("rMinPhi", false);
+  fMinPhiCmd->SetRange("rMinPhi>=0.&&rMinPhi<=360.");
+  fMinPhiCmd->SetDefaultUnit("rad");
```

Comment:
The range does not match the default unit, I suggest using deg as default.

Reply:
Oh, the default unit is rad so that the number we get internally is transformed to radians regardless of the input unit.

I think the issue is how to set the maximum to 2pi